connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -44,7 +45,7 @@ import org.apache.spark.sql.functions.lit
  *
  * @since 3.4.0
  */
-class Column private[sql] (private[sql] val expr: proto.Expression) {
+class Column private[sql] (private[sql] val expr: proto.Expression) extends Logging {
```

Comment:
Seems like we're not using this Logging

Reply:
The logging is needed for the binary compatibility: class type shall be exact the same.

Reply:
https://github.com/apache/spark/blob/97349986daa5aeacaa320f7187681a72c4830b13/sql/core/src/main/scala/org/apache/spark/sql/Column.scala#L142

should we delete `private[sql]` here?

Reply:
My poor Scala knowledge indicate this only marks one constructor private. The intension is to mark the current constructor private. For more constructor methods, we will add in follow up PRs.

Reply:
Hmm....why is it not consistent with `spark.sql.Column`?

Reply:
Our type is `proto.Expression`, it is not the same as `Expression`. I leave it to later PRs to decide how to support `Expression`.

Reply:
I mean, why not 

```
class Column(val expr: proto.Expression) extends Logging { ...
```

Reply:
Because I am not certain if we should expose constructor `this(expr: proto.Expression)` and `val expr:proto.Expression`.
They are not the same type as `this(expr: Expression)` and `val expr: Expression`. 

Our type `proto.Expression` is some grpc class and `Expression` is in sql package. They are different types from the binary code point of view.